### PR TITLE
[VarDumper] fix tests when xdebug is enabled

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20778
| License       | MIT

- Disabled some test cases when xdebug is enabled, see https://github.com/symfony/symfony/issues/20778

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php [hunk 1]
```diff
@@ -147,6 +147,10 @@ class: "Symfony\Component\VarDumper\Tests\Caster\ReflectionCasterTest"
      */
     public function testGenerator()
     {
+        if (extension_loaded('xdebug')) {
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php]: should be skipped only when ``xdebug.overload_var_dump`` is set to 2:

- 0 does not overload var_dump at all
- 1 overloads it only with formatting, which does not break our tests AFAIK
- 2 adds the location to the formatted output.

**Comment by stof** [comment]
note to mergers: this should go in older branches

File: src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php]: This is true, but for the next patch (this one is independent from var_dump)

**Comment by ShinDarth** [comment]
@stof done, changed the IF condition

File: src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php [hunk 2]
```diff
@@ -147,6 +147,10 @@ class: "Symfony\Component\VarDumper\Tests\Caster\ReflectionCasterTest"
      */
     public function testGenerator()
     {
+        if (ini_get('xdebug.overload_var_dump') == 2) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php]: sorry if the previous comment wasn't clear enough :)
this specific test should use extension_loaded as before (and the next check is now fine)

**Comment by ShinDarth** [comment]
@nicolas-grekas done, thanks

@stof can you please answer [my question](https://github.com/symfony/symfony/pull/20782#issuecomment-265156220) in that other PR?

**Comment by nicolas-grekas** [comment]
Thank you @ShinDarth.

**Pull request closed without merging**